### PR TITLE
Script to flash only if required

### PR DESCRIPTION
**Pull request by FranciscoCano**

### CODE REVIEW

File: scripts/flash_device.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import subprocess32
+import time
+from argparse import ArgumentParser
+import os
+import get_latest_build
+
+
+def connect_device():
```
Comment (villenavg) [inline scripts/flash_device.py]: We have this method in utilities. Why don't use it, instead. No cyclic dependencies afaik :D

File: scripts/flash_device.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    If the device has been already flashed with the latest available build,
+    return True. False otherwise.
+    """
+    output = subprocess32.check_output('sudo adb shell getprop | grep "build.date]"', shell=True)
+    build_date = output.split(": [")[-1].strip().replace("]", "")
+    da = time.strptime(build_date, "%a %b %d %H:%M:%S %Z %Y")
+    daf = time.strftime("%m-%d-%y", da)
+    print "Current build's date in device: {}".format(daf)
+    last_date = get_latest_build.detect_latest_date(source, user, passwd)
+    result = False
```
Comment (villenavg) [inline scripts/flash_device.py]: why dont just do like this: "return last_date == daf"

File: scripts/flash_device.sh [hunk 3]
```diff
@@ -89,35 +89,7 @@ fi
 
 cd $REL_DIR
 
-
```
Comment (villenavg) [inline scripts/flash_device.sh]: HELL YEAH

File: scripts/flash_device.py (hunk 1 above)
Comment (FranciscoCano) [inline scripts/flash_device.py]: You're right, I thought it was in the Toolkit, not available here. I'll change it.

File: scripts/flash_device.py (hunk 2 above)
Comment (FranciscoCano) [inline scripts/flash_device.py]: Doh!

File: scripts/flash_device.sh (hunk 3 above)
Comment (FranciscoCano) [inline scripts/flash_device.sh]: :dancer:

**Comment by villenavg** [comment]
LGTM

**Pull request merged**